nexus/src/app/mod.rs

```diff
@@ -245,6 +280,55 @@ impl Nexus {
         }
     }
 
+    // Called to hand off management of external servers to Nexus.
+    pub(crate) async fn set_servers(
+        &self,
+        external_servers: ExternalServers,
+        internal_server: DropshotServer,
+    ) {
+        // If any servers already exist, close them.
+        let _ = self.close_servers().await;
+
+        // Insert the new servers.
+        *self.external_servers.lock().await =
+            ExternalServers { ..external_servers };
```

Comment:
Nit - I think this could be

```suggestion
        *self.external_servers.lock().await = external_servers;
```

Reply:
Done in 70221549c05b500c2098efeb70619d874e94f62e